### PR TITLE
Add check clang-format workflow

### PR DESCRIPTION
**Pull request by sksat**
## Overview
- Add GitHub Actions workflow for check coding rule(ref: #33)

## Issue
- #33 

## Details
Run `clang-format` command with `git-clang-format`

##  Validation results
Link to tests or validation results.

## Scope of influence
eg. The behavior of XX will be change.

## Supplement
Write additional comments if you need.

## Note
- If there are related Projects, tie them together.
- Assignees should be set if possible.
- Reviewers should be set if possible.
- Set `priority` label if possible.

### CODE REVIEW

**Comment by sksat** [comment]
`git-clang-format`，in-placeオプションとか無いのか

File: .github/workflows/check-format.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - name: check clang-format version
+        run: |
+          clang-format --version
+          md5sum "$(which git-clang-format)"
+          head "$(which git-clang-format)"
+
+      - name: check format(push)
+        if: github.event_name == 'push'
+        run: |
+          git-clang-format --commit HEAD^ --diff | tee format.patch
```
Comment (meltingrabbit) [inline .github/workflows/check-format.yml:37]: diffがあれば落とす，なら最後の `name: check` いらなそう？って思ったけど，そうでもない？

File: .github/workflows/check-format.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      - name: check format(pull_request)
+        if: github.event_name == 'pull_request'
+        run: |
+          git-clang-format --commit ${{ github.event.pull_request.base.sha }} --diff | tee format.patch
+
+      - name: check
+        run: |
+          git apply --allow-empty format.patch
+          git status
+          git diff --exit-code
```
[the remark this answers is not in the document]
Comment (sksat) [reply .github/workflows/check-format.yml:48]: unstagedで表示はされるけどexit codeは変わらないですね．なので↓は通ってる．
https://github.com/ut-issl/s2e-core/runs/5558058408?check_suite_focus=true

**Comment by sksat** [comment]
approve貰ったらテストcommitだけ消してマージします

File: .github/workflows/check-format.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (meltingrabbit) [reply .github/workflows/check-format.yml:48]: おけ

**Review by meltingrabbit — APPROVED** [review]
よさそう

**Comment by sksat** [comment]
マージします

**Pull request merged**